### PR TITLE
Nspv json response buffer fix

### PR DESCRIPTION
**Pull request by dimxy**

### CODE REVIEW

File: src/komodo_nSPV_fullnode.h [hunk 1]
```diff
@@ -688,6 +688,7 @@ int32_t NSPV_remoterpc(struct NSPV_remoterpcresp *ptr,char *json,int n)
         {
             rpc_result = JSONRPCReplyObj(result, NullUniValue, jreq.id);
             response=rpc_result.write();
+            ptr->json = (char*)malloc(response.size());
```
Comment (jmjatlanta) [inline src/komodo_nSPV_fullnode.h:691]: I believe it possible that response.size() could be a large number. Should there be checks to assure that the malloc here (and below) did not return nullptr?

File: src/komodo_nSPV.h [hunk 2]
```diff
@@ -441,8 +441,10 @@ int32_t NSPV_rwremoterpcresp(int32_t rwflag,uint8_t *serialized,struct NSPV_remo
 
 void NSPV_remoterpc_purge(struct NSPV_remoterpcresp *ptr)
 {
-    if ( ptr != 0 )
+    if ( ptr != 0 )  {
```
[the remark this answers is not in the document]
Comment (dimxy) [reply src/komodo_nSPV.h:444]: I totally refactored the nspv code and replaced all such read/write functions to CDataStream object, sent it into research-new branch. So eventually all this code will be replaced

File: src/komodo_nSPV_fullnode.h (hunk 1 above)
Comment (dimxy) [reply src/komodo_nSPV_fullnode.h:691]: definitely should be checked, thank you

**Review by jmjatlanta — CHANGES_REQUESTED** [review]
Note that `NSPV_rwremoterpcresp` is never used to write to the struct, but would be dangerous if it tried to. I suggest either (a) renaming the method and removing the first parameter or (b) asserting if the first parameter is ever a 0.

Also, adding a constructor/destructor to the struct would move the need for the functionality of `memset` and `_purge` to a central location.

File: src/komodo_nSPV_defs.h [hunk 3]
```diff
@@ -186,7 +186,7 @@ struct NSPV_CCmtxinfo
 struct NSPV_remoterpcresp
 {
     char method[64];
-    char json[11000];
+    char *json;
```
[the remark this answers is not in the document]
Comment (dimxy) [reply src/komodo_nSPV_defs.h:190]: there is a memset to zero NSPV_remoterpcresp after the var decl 
but just added a constructor to auto init to zeros

**Comment by dimxy** [comment]
this PR is in the combined #559 PR

**Pull request closed without merging**